stdlib/src/array.rs

```diff
@@ -1258,9 +1258,11 @@ mod array {
 
     impl Iterable for PyArray {
         fn iter(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyResult {
+            let zelf_ = zelf.clone();
             Ok(PyArrayIter {
                 position: AtomicUsize::new(0),
                 array: zelf,
+                internal: PyMutex::new(PositionIterInternal::new(zelf_, 0)),
```

Comment:
```suggestion
                internal: PyMutex::new(PositionIterInternal::new(zelf.clone(), 0)),
```
Is `PositionIterInternal` necessary for implementing `__reduce__`?

Reply:
I think @Snowapril got confused here. because `zelf` is already moved in upper line, it can't be clone here.